src/main/java/io/cdap/plugin/servicenow/restapi/RestAPIClient.java

```diff
@@ -75,18 +89,15 @@ public RestAPIResponse executePost(RestAPIRequest request) throws IOException {
 
     try (CloseableHttpClient httpClient = HttpClientBuilder.create().build()) {
       try (CloseableHttpResponse httpResponse = httpClient.execute(httpPost)) {
+        validateHttpResponse(httpResponse);
         apiResponse = RestAPIResponse.parse(httpResponse, request.getResponseHeaders());
       }
     } catch (IOException e) {
       // We're retrying all transport exceptions while executing the HTTP POST method and the generic transport
       // exceptions in HttpClient are represented by the standard java.io.IOException class
       // https://hc.apache.org/httpclient-legacy/exception-handling.html
       throw e;
-    } catch (Exception e) {
```

Comment:
Although this is good for current use case, would this be backward incompatible with the existing pipeline?

Reply:
This is not backward, this exception is caught at a different layer within the connector.